gadgets/src/integers/int/int_impl.rs

```diff
@@ -29,6 +33,8 @@ macro_rules! int_impl {
 
         impl Integer for $name {
             type IntegerType = $type_;
+            type UnsignedGadgetType = $uname_;
+            type UnsignedIntegerType = $utype_;
```

Comment:
Explain: When programming the Div gadget for signed integers, now it uses the unsigned integers for help. This reduces the code complexity and seems to be the correct way to implement.